### PR TITLE
Added README files in v2 proto directories.

### PR DESCRIPTION
**Pull request by cody-littley**
## Why are these changes needed?

Adds README files in directories containing API not yet in use.

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by samlaf — COMMENTED** [review]
LGTM! thanks for this. would personally like to have a README in `api/proto/README.md` to describe v1 vs v2. Like an entrypoint README, that could link to the others.

File: api/proto/disperser/v2/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+No. The v2 API is currently under active development, and is not yet deployed to any environment.
+
+# Is the v2 API stable?
+
+No, although it is becoming more stable as the v2 design matures.
+
+# What is "v2"?
+
+The EigenDA v2 Architecture is a fundamental redesign of the disperser, and by extension systems that interact
+with the disperser. The v2 Architecture is an important stepping stone towards reducing centralization and for
```
Comment (ian-shim) [inline api/proto/disperser/v2/README.md:12]: nit: The v2 architecture improves robustness, efficiency, and compatibility with upcoming features such as permissionless disperser and data availability sampling.

File: api/proto/disperser/v2/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+No. The v2 API is currently under active development, and is not yet deployed to any environment.
+
+# Is the v2 API stable?
+
+No, although it is becoming more stable as the v2 design matures.
+
+# What is "v2"?
+
+The EigenDA v2 Architecture is a fundamental redesign of the disperser, and by extension systems that interact
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply api/proto/disperser/v2/README.md:11]: wording updated

File: api/proto/disperser/v2/README.md (hunk 1 above)
Comment (cody-littley) [reply api/proto/disperser/v2/README.md:12]: wording updated

**Comment by cody-littley** [comment]
@samlaf 

> LGTM! thanks for this. would personally like to have a README in api/proto/README.md to describe v1 vs v2. Like an entrypoint README, that could link to the others.

I added a readme to the `proto` directory. Not sure if it makes sense to link it to all of the other ones. The README files are basically all just copies of one another. They are currently spread around in multiple locations to make it more likely that somebody will think to look inside it when they are peeking into the v2 files.

File: api/proto/relay/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Should I use the v2 API?
```
Comment (ian-shim) [inline api/proto/relay/README.md:1]: There's no indication `api/proto/relay` is part of v2. Maybe this README should be specific to relay or say something like relay is only applicable with v2 architecture?

Comment (cody-littley) [reply api/proto/relay/README.md]: Added a note that the code in the `relay` directory is experimental.

**Comment by cody-littley** [comment]
> I feel like having one README at api/proto is sufficient, but I'll leave it up to you

I've collapsed these README files into a single file.

**Pull request merged**